src/layouts/directory.tsx

```diff
@@ -0,0 +1,68 @@
+import type {
+  DirectoryProfile,
+  LocationProfile,
+  TemplateRenderProps,
+} from "src/types/entities";
+import { AnalyticsScopeProvider } from "@yext/pages/components";
+import Breadcrumbs from "src/components/common/Breadcrumbs";
+import DirectoryCard from "src/components/cards/DirectoryCard";
+import DirectoryGrid from "src/components/directory/DirectoryGrid";
+import DirectoryHero from "src/components/directory/DirectoryHero";
+import DirectoryList from "src/components/directory/DirectoryList";
+
+interface DirectoryListLayoutProps {
+  data: TemplateRenderProps<DirectoryProfile<DirectoryProfile<never>>>;
+  directoryLevel: "root" | "region";
+}
+
+interface DirectoryGridLayoutProps {
+  data: TemplateRenderProps<DirectoryProfile<LocationProfile>>;
+  directoryLevel: "city";
+}
+
+type DirectoryLayoutProps = DirectoryListLayoutProps | DirectoryGridLayoutProps;
```

Comment:
These types are kind of complicated, but this works as expected with the types defined in each directory template file. Lmk if this is too complicated and I can split this into 3 different layout files.

Reply:
I don't love the directoryLevel prop. It feels a little arbitrary, what if someone wants to add a country level?

If its only use is to switch between the different templates maybe we could rename it to be more directly related to that functionality? Or could we potentially use the value of the dm_directoryChildren prop to detect which version to use automatically?